File: archives/2016-2017/lectures/14/Books/app/src/main/java/ro/ubbcluj/cs/books/NewBook.java
package ro.ubbcluj.cs.books;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.EditText;

import java.util.Date;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import ro.ubbcluj.cs.books.books.R;
import ro.ubbcluj.cs.books.domain.Book;

public class NewBook extends AppCompatActivity {

  @BindView(R.id.bookTitle)
  EditText bookTitle;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_new_book);
    ButterKnife.bind(this);
  }

  @OnClick(R.id.save)
  public void save(View view) {
    Manager manager = new Manager();
    manager.save(new Book(0, bookTitle.getText().toString(),new Date()));
    finish();
  }
}

File: lectures/14/Books/app/src/main/java/ro/ubbcluj/cs/books/NewBook.kt
package ro.ubbcluj.cs.books

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProviders
import kotlinx.android.synthetic.main.activity_new_book.*
import ro.ubbcluj.cs.books.domain.Book
import ro.ubbcluj.cs.books.model.MainModel
import java.util.*

class NewBook : AppCompatActivity() {
  private lateinit var model: MainModel

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_new_book)
    model = ViewModelProviders.of(this).get(MainModel::class.java)
    save.setOnClickListener {
      val app: BookApp = application as BookApp
      model.addBook(app, Book(0, bookTitle.text.toString(), Date()))
      finish()
    }
  }
}
